src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -127,16 +128,24 @@ public function catalogAction(
 
         // Set values from persistence and replace in the request
         $persistedFilterParameters = $productProvider->getPersistedFilterParameters();
-        /** @var FilterParametersUpdater $filterParametersUpdater */
-        $filterParametersUpdater = $this->get('prestashop.adapter.product.filter_parameters_updater');
-        $filters = $filterParametersUpdater->buildFilters(
+        /** @var ListParametersUpdater $listParametersUpdater */
+        $listParametersUpdater = $this->get('prestashop.adapter.product.list_parameters_updater');
+        $listParameters = $listParametersUpdater->buildListParameters(
             $request->query->all(),
             $persistedFilterParameters,
             compact('offset', 'limit', 'orderBy', 'sortOrder')
         );
-        extract($filters);
-
-        $persistedFilterParameters = array_replace($persistedFilterParameters, $request->request->all());
+        /*
+         * @var int $offset
+         * @var int $limit
+         * @var string $orderBy
+         * @var string $sortOrder
+         */
+        extract($listParameters);
```

Comment:
Are you sure you control everything about data here?
```
Warning Do not use extract() on untrusted data, like user input (e.g. $_GET, $_FILES).
```

Reply:
well the `ListParametersUpdater` kind of merges the params from query, persisted filters, and default filters, and the return them as an array

so a query parameter can be returned in the array (which is formatted by the class), in which case could it be a problem? because it doesn't change much from getting the parameter from the request directly

the important being, it does not extract any/all variables from $_GET but a specific array composed internally: https://github.com/PrestaShop/PrestaShop/blob/25a55e3eb7668ffc4310be81637a3e308da7a098/src/Adapter/Product/ListParametersUpdater.php#L68

Reply:
> The best option is to not use extract() at all. It's a bad design decision from the days when PHP was the equivalent of wet toilet paper for writing secure code.

> Working with arrays in most languages is far better because the compiler and/or interpreter can use SIMD instructions with it. (https://en.wikipedia.org/wiki/SIMD)

What happened if `ListparametersUpdater` change, you can not anticipate what variables you will create, magic is a bad practice especially in PHP environment :/ 
More extract is slower than a foreach.

Reply:
so I should extract them manually?

```php
$offset = $listParameters['offset'];
$limit = $listParameters['limit'];
$orderBy = $listParameters['orderBy'];
$sortOrder = $listParameters['sortOrder'];
```

or replace the subsequent calls in the controller using the array?

```php
$products = $productProvider->getCatalogProductList($listParameters['offset'], $listParameters['limit'], $listParameters['orderBy'], $listParameters['sortOrder'], $request->request->all());
```

there is also the possibility to create a `ListParameters` object

Reply:
Depends if you need to reuse variables after otherwise the second solution can be fine.

Reply:
yep they are injected in the view later

Reply:
I sure can override them manually as in the first example, I just wanted to simplify the code, but if it's not clear enough (extract hides a lot of stuff) maybe it's useless
That's actually why I added the phpdoc to know which variables are extracted, which kills the gain of lines